### PR TITLE
Create jupyterlab execute time

### PR DESCRIPTION
**Pull request by vglfr**
## Description of changes

Jupyterlab extension to display cell execution time https://github.com/deshaw/jupyterlab-execute-time.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/jupyterlab-execute-time/default.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    hash = "sha256-DiyGsoNXXh+ieMfpSrA6A/5c0ftNV9Ygs9Tl2/VEdbk=";
+  };
+
+  # jupyterlab is required to build from source but we use the pre-build package
+  postPatch = ''
+    substituteInPlace pyproject.toml \
+      --replace '"jupyterlab~=4.0.0"' ""
+  '';
+
+  nativeBuildInputs = [
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/jupyterlab-execute-time/default.nix:25]:
```suggestion
  dependencies = [
```

File: pkgs/development/python-modules/jupyterlab-execute-time/default.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    jupyter-packaging
+  ];
+
+  # has no tests
+  doCheck = false;
+
+  pythonImportsCheck = [ "jupyterlab_execute_time" ];
+
+  meta = with lib; {
+    description = "A JupyterLab extension for displaying cell timings";
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/jupyterlab-execute-time/default.nix:36]:
```suggestion
    description = "JupyterLab extension for displaying cell timings";
```

File: pkgs/development/python-modules/jupyterlab-execute-time/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    jupyterlab
+    jupyter-packaging
+  ];
+
+  # has no tests
+  doCheck = false;
+
+  pythonImportsCheck = [ "jupyterlab_execute_time" ];
+
+  meta = with lib; {
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/jupyterlab-execute-time/default.nix:35]: Better to avoid `with lib;`

**Comment by vglfr** [comment]
Fixed.

**Comment by vglfr** [comment]
@NixOS/nixpkgs-merge-bot merge

**Pull request merged**